src/main/java/org/broadinstitute/hellbender/engine/dataflow/transforms/RemoveDuplicateReadVariantPairs.java

```diff
@@ -0,0 +1,152 @@
+package org.broadinstitute.hellbender.engine.dataflow.transforms;
+
+import com.google.api.client.util.Sets;
+import com.google.cloud.dataflow.sdk.transforms.*;
+import com.google.cloud.dataflow.sdk.transforms.join.CoGbkResult;
+import com.google.cloud.dataflow.sdk.transforms.join.CoGroupByKey;
+import com.google.cloud.dataflow.sdk.transforms.join.KeyedPCollectionTuple;
+import com.google.cloud.dataflow.sdk.values.*;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+
+import java.util.*;
+
+
+/**
+ * RemoveDuplicateReadVariantPairs uses the UUID surrogates for GATKRead and Variant to remove duplicate KV pair values.
+ * This currently requires 9 transforms.
+ * (1) pTransform to get PCollection<KV<UUID, UUID>>
+ * (2) remove dupes of (1)
+ * (3) GroupByKey of (2) (produces PCollection<UUID,Iterable<UUID>>)
+ * (4) create PCollection<UUID, KV<UUID, Variant>>
+ * (5) create KeyedPCollectionTuple of (3) and (4)
+ * (6) use (5) to create PCollection<UUID, Iterable<Variant>>
+ * (7) use  pKvAB to create PCollection<KV<UUID, GATKRead>>
+ * (8) create KeyedPCollectionTuple of (6) and (7)
+ * (9) join Alpha back in (using (8)) to get PCollection<GATKRead, Iterable<Variant>>
```

Comment:
Is there a reason you're doing it like this instead of:

1. add a key that's the UUID of the read and variant 
2. Group.perKey, using "first" as the grouping function
3. remove the key